### PR TITLE
Log outgoing HTTP path parameters in Structured mode

### LINKED ISSUE #4698: HttpClient logging doesn't log path parameters in Structured mode
Consider this example:
```cs
var builder = Host.CreateApplicationBuilder(args);

builder.Logging.AddJsonConsole();

builder.Services.AddRedaction();

builder.Services.AddExtendedHttpClientLogging(options =>
{
    options.RequestPathLoggingMode = OutgoingPathLoggingMode.Structured;
    options.RequestPathParameterRedactionMode = HttpRouteParameterRedactionMode.Loose;
});

var host = builder.Build();

var httpClientFactory = host.Services.GetRequiredService<IHttpClientFactory>();
using var httpClient = httpClientFactory.CreateClient();

using var requestMessage = new HttpRequestMessage(HttpMethod.Get, "https://api.github.com/users/aspnet/repos");
var requestMetadata = new RequestMetadata()
{
    RequestRoute = "users/{userId}/repos",
    DependencyName = "GitHub",
    RequestName = "Get user repos"
};

requestMessage.SetRequestMetadata(requestMetadata);
using var response = await httpClient.SendAsync(requestMessage);
```

That way you don't have a chance to see the value of `userId` path parameter in logs. The same `RequestPathLoggingMode` applied in incoming HTTP logging emits path parameters as separate tags.
Is it a bug?

### PR DESCRIPTION
**Pull request by xakep139**
Fixes #4698

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/extensions/pull/4845)

### CODE REVIEW

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/Log.cs [hunk 1]
```diff
@@ -171,7 +171,10 @@ public static void EnrichmentError(ILogger logger, Exception exception, string?
         var requestHeadersCount = record.RequestHeaders?.Count ?? 0;
         var responseHeadersCount = record.ResponseHeaders?.Count ?? 0;
 
-        var index = loggerMessageState.ReserveTagSpace(MinimalPropertyCount + statusCodePropertyCount + requestHeadersCount + responseHeadersCount);
+        var spaceToReserve = MinimalPropertyCount + statusCodePropertyCount + requestHeadersCount + responseHeadersCount +
+            record.PathParametersCount + (record.RequestBody is null ? 0 : 1) + (record.ResponseBody is null ? 0 : 1);
```
Comment (geeknoid) [inline src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/Log.cs:175]: Now that I've added support for the [TagName] property, can we get rid of this custom code and use the code generator instead?

**Review by geeknoid — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/Log.cs (hunk 1 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/Log.cs:175]: I have doubts regarding prefixes in header names, but I will try to convert this logic

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/LoggerMessageStateExtensions.cs [hunk 2]
```diff
@@ -26,8 +27,7 @@ public static void AddRequestHeaders(this LoggerMessageState state, List<KeyValu
         {
             var key = _requestPrefixedNamesCache.GetOrAdd(
                 items[i].Key,
-                static (x, p) => p + Normalize(x),
-                HttpClientLoggingTagNames.RequestHeaderPrefix);
+                static x => HttpClientLoggingTagNames.RequestHeaderPrefix + Normalize(x));
```
Comment (iliar-turdushev) [inline src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/LoggerMessageStateExtensions.cs:30]: Will this transformation lead to a delegate allocation on each invocation of GetOrAdd?

**Review by iliar-turdushev — APPROVED** [review]

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/LoggerMessageStateExtensions.cs (hunk 2 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/LoggerMessageStateExtensions.cs:30]: No, the compiler is smart enough to create a `static` backing field and assign the delegate instance to it just once:
```cs
string key = LoggerMessageStateExtensions._requestPrefixedNamesCache.GetOrAdd(items[i].Key, LoggerMessageStateExtensions.<>c.<>9__2_0 ?? (LoggerMessageStateExtensions.<>c.<>9__2_0 = new Func<string, string>((object) LoggerMessageStateExtensions.<>c.<>9, __methodptr(<AddRequestHeaders>b__2_0))));

//...
    [CompilerGenerated]
    [Serializable]
    private sealed class <>c
    {
      [Nullable(0)]
      public static readonly LoggerMessageStateExtensions.<>c <>9;
      [Nullable(new byte[] {0, 1, 1})]
      public static Func<string, string> <>9__2_0;
      // ...
    }
```

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/Microsoft.Extensions.Http.Diagnostics.csproj [hunk 3]
```diff
@@ -42,6 +42,7 @@
     <PackageReference Include="Microsoft.Extensions.Http" />
     <PackageReference Include="System.Text.Json" />
     <PackageReference Include="System.Collections.Immutable" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net8.0'))" />
+    <PackageReference Include="System.Net.Http" Condition="'$(TargetFramework)' == 'net462'" />
```
Comment (xakep139) [inline src/Libraries/Microsoft.Extensions.Http.Diagnostics/Microsoft.Extensions.Http.Diagnostics.csproj:45]: This is to update the implicit reference of 4.3.0 which is vulnerable

File: src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/Log.cs (hunk 1 above)
Comment (xakep139) [reply src/Libraries/Microsoft.Extensions.Http.Diagnostics/Logging/Internal/Log.cs:175]: Done in #4842

**Pull request merged**